### DISCUSSION TITLE
Spherical Treadmill: air cushion vs. ball transfer bearings

### DISCUSSION THREAD

**Discussion by emikrish** [sn-lab/MouseGoggles#16] [General]
Hello!  This repository suggests using an air-cushioned spherical treadmill as used in [Harvey 2009](https://pubmed.ncbi.nlm.nih.gov/19829374/) and many other experiments after that.    I was wondering if you have any thoughts about using [nylon/plastic ball transfer bearings](https://www.amazon.com/uxcell-Transfer-Bearing-Bolt-Fixing-Transmission/dp/B08LCY4LFW/ref=sr_1_6?crid=38HDSMDO367NV&keywords=plastic%2Bball%2Btransfer%2Bbearings&qid=1690395594&sprefix=plastic%2Bball%2Btransfer%2Bbearings%2Caps%2C78&sr=8-6&th=1) instead of an air cushion, like the setup used in [Miguelez Fernández 2018](https://www.sciencedirect.com/science/article/pii/S0165027017304363), which proposes a treadmill setup for rats (here's a [video](https://www.sciencedirect.com/science/article/pii/S0165027017304363#sec0115)).  It's quieter for sure and easier to set up, but it's definitely not 'frictionless' -- I'm wondering if it's low-friction enough for a mouse (much lighter and weaker than a rat) to easily navigate. We might try it out, but I was wondering if you had given any thought to it already.

**Comment by misaacson01** [comment]
We've considered tons of treadmill ideas but actually not that specific one. I could see this working for mice, with really nice, well-maintained bearings, but I don't know for sure. Compared to the other options out there (e.g. linear or toroidal treadmills, mobile home cage, slippery floors) this idea is overall not too different. And I've come to the conclusion that there are no perfect solutions -- mice can learn to walk on treadmills easy enough but turning and maintaining virtual headings is too challenging -- so you just gotta pick one that works OK. Air-supported can be pretty quiet and simple if you have a building with air supply, but if you need to buy a separate pump then they can be really loud. 

If you give the bearings a try I'll be curious to hear how mice handle it!

**Comment by ZhaomingPsych** [comment]
Hi,[monrky](https://github.com/monrky). Instead of using an air cushion ball, we have implemented a treadmill to build up VR system, drawing inspiration from [Rune Nguyen Rasmussen's work](https://github.com/Neurune/TreadmillSystem). The system is straightforward to put together and operates exceptionally well. 
![f7672f16eab3d337fac1e19081e9b0f](https://github.com/sn-lab/mouseVRheadset/assets/96122605/341ba96c-bb86-49ec-b333-8baa851984ff)

https://github.com/sn-lab/mouseVRheadset/assets/96122605/685995f6-c021-42b3-8933-9b1434b41e50

**Reply by misaacson01** [reply]
Along the same lines, we've tested the [Janelia low-friction linear treadmill](https://www.labmaker.org/products/mouse-treadmill-with-encoder) for simple 1D VR tasks, which also works great.

**Reply by emikrish** [reply]
Hi @misaacson01 -- hope you are doing well!  We actually just purchased this exact treadmill. I am working on integrating it with the mouseGoggles/Godot. Should I upload your ([sketch](https://github.com/sn-lab/MouseGoggles/blob/main/Other%20Hardware/Linear%20Treadmill/LinearTreadmill_Mouse_Controller_V2/LinearTreadmill_Mouse_Controller_V2.ino) to the encoder that came with the treadmill (i.e. overwrite the existing programming), or attach the encoder to an arduino running your sketch?

**Reply by misaacson01** [reply]
Hi @monrky! You'll have to upload that sketch directly to the treadmill's microcontroller. Programming "Teensy" boards isn't always a perfectly smooth experience, but if you follow the [instructions](https://github.com/sn-lab/MouseGoggles/blob/main/Other%20Hardware/Installation%20README.md#linear-treadmill) to the letter it should work on the first try. If something in the instructions doesn't work, let me know, I've had to change the instructions a couple times after big Arduino IDE updates.

After you program it, you can check whether it worked by moving the treadmill by hand -- when plugged into any PC, it should make the mouse cursor move up and down. Let me know how it goes!

**Reply by emikrish** [reply]
got it!! it worked. thanks!!
